.eslintrc.yml

```
@@ -0,0 +1,89 @@
+env:
+  browser: true
+  node: true
+  es2021: true
+
+ignorePatterns:
+  - 'public/'
+  - '.cache/'
+  - 'src/pages/**/*.mdx'
+
+extends:
+  - eslint:recommended
+  - plugin:import/recommended
+  - plugin:import/typescript
+  - plugin:@typescript-eslint/recommended
+  - plugin:prettier/recommended
+
+overrides:
+  - files:
+    - 'gatsby-config.ts'
+    rules:
+      '@typescript-eslint/quotes':
+        - off
+  - files:
+    - '*.json'
+    rules:
+      '@typescript-eslint/quotes':
+        - off
+      '@typescript-eslint/comma-dangle':
+        - off
+      '@typescript-eslint/semi':
+        - off 
+
+parser: "@typescript-eslint/parser"
+
+rules:
+  "react/prop-types":
+    - "off"
+  no-multiple-empty-lines:
+    - error
+    - max: 1
+  no-var:
+    - error
+  no-console:
+    - error
+  arrow-parens:
+    - error
+    - always
+  curly:
+    - error
+    - all
+  no-else-return:
+    - error
+    - allowElseIf: false
+  no-duplicate-imports:
+    - error
+    - includeExports: true
+  jsx-quotes:
+    - error
+    - prefer-double
+  '@typescript-eslint/explicit-function-return-type':
+    - error
+    - allowTypedFunctionExpressions: true
+  '@typescript-eslint/quotes':
+    - error
+    - single
+  '@typescript-eslint/object-curly-spacing':
+    - error
+    - always
+  '@typescript-eslint/indent':
+    - off
+  '@typescript-eslint/semi':
+    - error
+    - always
+  '@typescript-eslint/comma-dangle':
+    - error
+    - always-multiline
+  '@typescript-eslint/no-use-before-define':
+    - error
+  '@typescript-eslint/no-unused-vars':
+    - error
+    - vars: all
+      args: after-used
+      argsIgnorePattern: ^_
+      caughtErrors: all
+  import/no-unresolved:
+    - off
+  import/no-default-export:
+    - off
```

there should not be default exports elsewhere except  gatsby-config.ts, src/index.d.ts, src/pages/**/*.tsx, override this rule only for these files
```suggestion 
  import/no-default-export:
    - error
```

---

Gatsby framework requires default exports.

---

https://github.com/B1naryStudio/bsa-website/blob/d4fab674591d74c06914f51a440591d731e3e52b/blog/.eslintrc.yml#L18

---

Turn off this rule only for specific files